test/extended/builds/gitauth.go

```diff
@@ -0,0 +1,93 @@
+package builds
+
+import (
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	testutil "github.com/openshift/origin/test/util"
+)
+
+// hostName removes a port from a host specification
+func hostName(host string) string {
+	parts := strings.Split(host, ":")
+	return parts[0]
```

Comment:
This should use net.SplitHostPort and handle cases where no port was part of the host string